plugins/techdocs/src/reader/components/Reader.tsx

```diff
@@ -31,10 +28,13 @@ import transformer, {
 } from '../transformers';
 import { docStorageURL } from '../../config';
 import URLFormatter from '../urlFormatter';
+import { TechDocsNotFound } from './TechDocsNotFound';
+import { TechDocsPageWrapper } from './TechDocsPageWrapper';
 
 const useFetch = (url: string) => {
   const state = useAsync(async () => {
     const response = await fetch(url);
+    if (response.status === 404) return '404';
```

Comment:
Could we not do something like this:

```suggestion
    if (response.status === 404) {
      return new Error('Page not found');    
    }
```

And the later we can do

```js
if (state.value instanceof Error) {
  return <TechDocsNotFound />;
  // Possibly even this?
  // return <TechDocsNotFound errorMessage={state.value.message} />
}
```